specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json

```diff
@@ -0,0 +1,1534 @@
+            "Standard_D11_v2",
+            "Standard_D12_v2",
+            "Standard_L4s",
+            "Dev(No SLA)_Standard_D11_v2",
+            "Standard_E64i_v3",
+            "Standard_E80ids_v4",
+            "Standard_E2a_v4",
+            "Standard_E4a_v4",
+            "Standard_E8a_v4",
+            "Standard_E16a_v4",
+            "Standard_E8as_v4+1TB_PS",
+            "Standard_E8as_v4+2TB_PS",
+            "Standard_E16as_v4+3TB_PS",
+            "Standard_E16as_v4+4TB_PS",
+            "Dev(No SLA)_Standard_E2a_v4"
+          ],
+          "x-ms-enum": {
+            "name": "AzureSkuName",
+            "modelAsString": true
+          }
+        },
+        "capacity": {
+          "description": "The number of instances of the cluster.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "tier": {
+          "description": "SKU tier.",
+          "type": "string",
+          "enum": [
+            "Basic",
+            "Standard"
+          ],
+          "x-ms-enum": {
+            "name": "AzureSkuTier",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Azure SKU definition."
+    },
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody",
+          "description": "An error response from Kusto."
+        }
+      },
+      "description": "An error response from Kusto."
+    },
+    "CloudErrorBody": {
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for displaying in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from Kusto."
+    },
+    "KustoPoolProperties": {
+      "properties": {
+        "state": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The state of the resource.",
+          "enum": [
+            "Creating",
+            "Unavailable",
+            "Running",
+            "Deleting",
+            "Deleted",
+            "Stopping",
+            "Stopped",
+            "Starting",
+            "Updating"
+          ],
+          "x-ms-enum": {
+            "name": "State",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
```

Comment:
How is the `state` property will be different than `provisioningState`? What value would `provisioningState` return? Please follow provisioning state semantics document: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property)

Reply:
The _state_ property reflects specific intermediate states of Kusto Pool/Cluster to the customer.
The _provisioningState_ property reflects general state in which the Kusto Pool/Cluster is in.
For example, the _state_ can be "Updating" or "Starting" while the _provisioningState_ is "Running".

It's another layer provided to the customer to reflect KustoPool complete state overview.

Reply:
I see - in that case can you update the `provisioningState` property to be an enum with list of provisioning states. Maybe use refer to ResourceProvisioningState model below?

---
In reply to: [612471921](https://github.com/Azure/azure-rest-api-specs/pull/13874#discussion_r612471921) [](ancestors = 612471921)

Reply:
Sure, changed.